### DISCUSSION TITLE
Mean Variance Portfolios with Custom Parameters

### DISCUSSION THREAD

**Discussion by sirhc78** [dcajasn/Riskfolio-Lib#78] [General]
Hi, congratulations for the work and your impressive Riskfolio-Lib.

I would like to know if there is a way to take in account some custom skew and kurtosis.
Sometimes, qualitative analysis permits to better estimate a skew. I would like to force a worst skew.
As we can input mu and cov, maybe this is not related to the lib, and I should just search on how compute a covariance matrix with predefined skew and kurtosis.
Do you know how could I do that ?
Your experience and thoughts would be greatly appreciated.

**Comment by dcajasn** [comment]
Hi @sirhc78 

The problem with skewness is that it can't be modeled using convex optimization techniques because is grade 3 so the function is not convex or concave. Kurtosis can be modeled because is grade 4 but is very difficult because you need to use sum of squares. Here is a [paper](https://palomar.home.ece.ust.hk/papers/2021/ZhouPalomar-TSP2021.pdf) of Daniel Palomar that uses a technique called majorization minimization to optimize skewness and kurtosis. Konno uses milp to approximante skewness in this [paper](https://www.worldscientific.com/doi/abs/10.1142/S0219024905003116).

Best,
Dany

**Reply by sirhc78** [reply]
Thank you. You are right.

So I should consider using constraints that incorporate an expected skewness or max drawdown  for each underlying assets in portfolio. Do you have anything like this?

**Reply by dcajasn** [reply]
Riskfolio-lib allows drawdown contraints but not skewness because it can't be modeled using convex formulations. For skewness you only can get a local optimal using an approximation like in the papers I told you.
